### PR TITLE
Use Zeroize Crate to scrub memory.

### LINKED ISSUE #843: slice::fill(0) Does Not Always Zero Memory
Memory that is not zeroed stays resident for longer than strictly necessary, potentially
allowing it to be exposed by a separate memory exfiltration vulnerability.

### PR DESCRIPTION
**Pull request by mhatrevi**
This PR addresses issue https://github.com/chipsalliance/caliptra-sw/issues/843

### CODE REVIEW

File: rom/dev/src/flow/cold_reset/crypto.rs [hunk 1]
```diff
@@ -36,21 +37,16 @@ impl Ecdsa384SignatureAdapter for Ecc384Signature {
 }
 
 /// DICE  Layer Key Pair
-#[derive(Debug)]
+#[derive(Debug, Zeroize)]
```
Comment (bluegate010) [inline rom/dev/src/flow/cold_reset/crypto.rs:40]: Should these be `ZeroizeOnDrop`?

Comment (mhatrevi) [reply rom/dev/src/flow/cold_reset/crypto.rs:40]: We are explicitly calling the zeroize trait method on the critical structs, so just having Zeroize is good.

Comment (korran) [reply rom/dev/src/flow/cold_reset/crypto.rs:40]: It's worth considering in a future PR; I suspect we end up forgetting to zeroize many of these.

**Review by korran — APPROVED** [review]

File: rom/dev/src/flow/cold_reset/crypto.rs (hunk 1 above)
Comment (mhatrevi) [reply rom/dev/src/flow/cold_reset/crypto.rs:40]: Sounds good.

**Review by jhand2 — APPROVED** [review]

**Pull request merged**